### ISSUE TITLE
Make code intelligence availability and status transparent to users

### ISSUE THREAD

**Issue by macraig** [sourcegraph/sourcegraph#32293]
#### Feature request description

Add transparency to code intelligence UX by telling users when precise or search-based code intel is available, what data is used and how old that data is. 

#### Is your feature request related to a problem?

Users don't know what to expect or why things happened the way they did. They're unsure whether s popover didn't show up because of network problems, or because there never will be popover on that token (some tokens don't have popovers and that's completely fine), or because no data is available (search-based is indexing in the background, no LSIF dump uploaded), etc.

**Comment by mrnugget** [comment]
How does this ticket relate to https://github.com/sourcegraph/sourcegraph/issues/32294?

**Comment by jjinnii** [comment]
@mrnugget agreed, I think we need to break this issue down into what user problem it should tackle to understand how it relates to the other issues in #32294 (as to not be duplicative). 

These are a few user pain points that "code intel availability and status" _could_ cover (although not necessarily should). This is lengthy, so we can discuss more in planning tomorrow. 

- Is code intelligence available for this token?
  - I hovered over a token, saw a blank popover with spinner, then it disappeared. Is intel available or is SG not working? 
    - #19034 will address this behavior happening due to there not being a definition available.
    - @macraig Could you explain more in detail what network problems means and how that currently manifests in the UI? 
  - I expected there to be intel available for this token, but nothing is coming up. Why? 
    - Do we have any documented types of tokens that we lack any code intel support for that users expect? (e.g. a gap in our code intelligence) 
- Is code intelligence available for the file? 
  - I think it's important to understand the problems ^ in the prior point, before jumping to "users need a way to see what we offer code intel on before hovering"
- I was expecting precise intel, why am I seeing search-based? 
  - Aside from the spinner situation, do we have documented feedback (internal/external) of users wanting to understand/not understanding why we fallback to search-based?
- Why is this code intelligence so inaccurate?
  - Is it even possible to identify where our search-based intel may not be that great? 🤔

**Comment by mrnugget** [comment]
> @mrnugget agreed, I think we need to break this issue down into what user problem it should tackle to understand how it relates to the other issues in #32294 (as to not be duplicative).

I think you might have understood my question as more nuanced than it really was 😄 I really meant: isn't this ticket here a duplicate of https://github.com/sourcegraph/sourcegraph/issues/32294? #32294 includes several other tickets, including #8830, which explicitly mentions one of the first problems you mentioned under "Is code intelligence available for this token?"

In [this Slack thread](https://sourcegraph.slack.com/archives/CHXHX7XAS/p1646648630001659) I posted two recordings of this happening

> I hovered over a token, saw a blank popover with spinner, then it disappeared. Is intel available or is SG not working?

which is https://github.com/sourcegraph/sourcegraph/issues/19034.

And in that thread [@macraig links](https://sourcegraph.slack.com/archives/CHXHX7XAS/p1646743377005749?thread_ts=1646648630.001659&cid=CHXHX7XAS) to #8830.

So that tells me #8830 and #19034 are related (the former describes two problems, one of them being the latter) and both of these are included in #32294.

So can't we merge this ticket here with #32294 at least until we have a clear distinction and know that we need two tickets?

Because I have to confess it's really hard for me to keep track of all of these issues and what bit of information is in which ticket and which ticket is solving what exactly.

**Comment by Numbers88s** [comment]
Closed this one. Continuing discussion in https://github.com/sourcegraph/sourcegraph/issues/32294
